packages/dashboard-hardhat-plugin/src/index.ts

```diff
@@ -0,0 +1,59 @@
+import { task } from "hardhat/config";
+import { ActionType } from "hardhat/types";
+
+const FromHardhat = require("@truffle/from-hardhat");
+
+const {
+  DashboardMessageBusClient
+} = require("@truffle/dashboard-message-bus-client");
+
+interface CompilationArgs {}
+
+const processCompilation: ActionType<CompilationArgs> = async (
+  taskArgs,
+  env,
+  runSuper
+) => {
+  // Run Hardhat compilation
+  const result = await runSuper();
+
+  try {
+    await FromHardhat.expectHardhat();
+
+    const dashboardConfig = {
+      host: "localhost",
+      port: 24012
+    };
+
+    console.log("Preparing Truffle compilation");
+
+    // Prepare Truffle compilations
+    let compilations = await FromHardhat.prepareCompilations();
+
+    // Create Message Bus client
+    let messageBusClient = new DashboardMessageBusClient(dashboardConfig);
+
+    console.log("Sending Truffle compilation to dashboard");
+
+    // Send Truffle compilations to dashboard
+    const publishLifecycle = await messageBusClient.publish({
+      type: "cli-event",
+      payload: {
+        label: "workflow-compile-result",
+        data: { compilations }
+      }
+    });
+    publishLifecycle.abandon();
+
+    console.log("Compilation successfully sent!");
+  } catch (hardhatError) {
+    console.log("Error: ", hardhatError);
+    throw hardhatError;
```

Comment:
Do we want this to fail? Maybe the plugin should only throw a warning and return, so that other plugins afterwards can still function.

Reply:
Removed the throw and tweaked the accompanying message in fa8deb165ad3653dc394573b70e9700369bd90fd. Feel free to review + comment / resolve.